src/deploy/NVA_build/noobaa_init.sh

```diff
@@ -76,7 +69,65 @@ extract_noobaa_in_docker() {
   fi
 }
 
-run_agent_container() {
+# run_internal_process runs a process and handles NOOBAA_INIT_MODE.
+#
+# NOOBAA_INIT_MODE allows devs to set how the container behaves when the process exits.
+# Possible mode values are:
+# - "" (no env/file or empty string) is the default and means to exit the container
+# - "auto" restart the process without letting the container die to allow reloading code.
+# - "manual" the container will loop and wait for manual intervention to change the mode.
+#
+# Usage: set env NOOBAA_INIT_MODE on the deployment pod spec
+#   or write the value to /root/node_modules/noobaa-core/NOOBAA_INIT_MODE
+#   (the file will override the env).
+#
+run_internal_process() {
+  while true
+  do
+    echo "Running: $*"
+    $*
+    rc=$?
+    echo -e "\n\n\n"
+    echo "######################################################################"
+    echo "$(date) NooBaa: Process exited RIP (RC=$rc)"
+    echo "######################################################################"
+    echo -e "\n\n\n"
+
+    mode="manual" # initial value just to start the loop
+    while [ "$mode" == "manual" ]
+    do
+      # load mode from file/env
+      if [ -f "./NOOBAA_INIT_MODE" ]
+      then
+        mode="$(cat ./NOOBAA_INIT_MODE)"
+      else
+        mode="$NOOBAA_INIT_MODE"
+      fi
+
```

Comment:
We can Add here `mode=${mode,,}` to make sure the string is in lowercase

Reply:
Cool, or maybe not... 
This is quite a cryptic syntax for lowercasing a string, no?
Are you really worried about case mistakes or just having some fun with bash? :)

Reply:
some fun with bash 😂  
and due to the fact that one needs to know what to put there as env, this is not critical

Reply:
https://www.cyberciti.biz/faq/linux-unix-shell-programming-converting-lowercase-uppercase/